libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/ShareDialogBase.tsx

```diff
@@ -0,0 +1,120 @@
+// (C) 2021 GoodData Corporation
+import React, { useCallback, useMemo, useState } from "react";
+import { areObjRefsEqual, uriRef } from "@gooddata/sdk-model";
+import { Overlay } from "../../../Overlay";
+import { IAlignPoint } from "../../../typings/positioning";
+import { ShareGranteeBase } from "./ShareGranteeBase";
+import { AddGranteeBase } from "./AddGranteeBase";
+import { DialogModeType, GranteeItem, IShareDialogBaseProps } from "./types";
+import { notInArrayFilter, GROUP_ALL_ID } from "./utils";
+
+const alignPoints: IAlignPoint[] = [{ align: "cc cc" }];
+
+const availableGranteesConst: GranteeItem[] = [
+    {
+        id: uriRef(GROUP_ALL_ID),
+        type: "groupAll",
+    },
+];
+
+/**
+ * @internal
+ */
+export const ShareDialogBase: React.FC<IShareDialogBaseProps> = (props) => {
```

Comment:
this is getting long, how about custom hook to wrap all the react hook stuff?

Reply:
In next story more hooks will be added so I will refactor this.